pkgs/development/cuda-modules/setup-hooks/mark-for-cudatoolkit-root/mark-for-cudatoolkit-root.sh

```diff
@@ -0,0 +1,78 @@
+# shellcheck shell=bash
+
+# shellcheck disable=SC1091
+source @logFromSetupHook@
+
+# Guard helper function
+# Returns 0 (success) if the hook should be run, 1 (failure) otherwise.
+# This allows us to use short-circuit evaluation to avoid running the hook when it shouldn't be.
+markForCUDAToolkit_ROOTGuard() {
+    local guard="Skipping"
+    local reason=""
+
+    # This hook is meant only to add a stub file to the nix-support directory of the package including it in its
+    # nativeBuildInputs, so that the setup hook propagated by cuda_nvcc, setup-cuda, can detect it and add the
+    # package to the CUDA toolkit root. Therefore, since it only modifies the package being built and will not be
+    # propagated, it should only ever be included in nativeBuildInputs.
+    if (( ${hostOffset:?} == -1 && ${targetOffset:?} == 0)); then
+        guard="Sourcing"
+        reason="because the hook is in nativeBuildInputs relative to the package being built"
+    fi
+
+    logFromSetupHook \
+        "INFO" \
+        "mark-for-cudatoolkit-root" \
+        "markForCUDAToolkit_ROOTGuard" \
+        "${hostOffset:?}" \
+        "${targetOffset:?}" \
+        "$guard $reason"
+
+    # Recall that test commands return 0 for success and 1 for failure.
+    [[ "$guard" == Sourcing ]]
+    return $?
+}
+
+# Guard against calling the hook at the wrong time.
+markForCUDAToolkit_ROOTGuard || return 0
+
+# Make a copy of the current offsets, so that we can use them in information messages; this is necessary because the
+# offsets are not consistently available in the environment during various phases of the build.
+declare -g snapshotHostOffset="${hostOffset:?}"
+declare -g snapshotTargetOffset="${targetOffset:?}"
```

Comment:
I've used `-g` in several places as well and I'm quite hazy on what the exact implications are and whether that was really necesasry

Reply:
Yes, this was a bad copy-paste. The global flag (`-g`) is only necessary if you want to introduce a variable into the global scope from within a function body. Since these are declared outside of a function, they're already in the global scope :l
Will be resolved in latest.